client.go

```diff
@@ -36,34 +48,77 @@ func NewClient(c Config) (*Client, error) {
 		return nil, errors.New("password is missing")
 	}
 
+	if c.TTL == 0 {
+		return nil, errors.New("zero is not a valid ttl")
+	}
+
 	return &Client{
-		client: goredis.NewClient(&goredis.Options{
+		client: RedisClient{client: redis.NewClient(&redis.Options{
 			Addr:     c.Addr,
 			Password: c.Password,
 			DB:       c.Database,
-		}),
+		})},
 		ttl: c.TTL,
 	}, nil
 }
 
-//Set add session to redis
-func (c *Client) Set(s *session.Session) error {
+// Set - add session to redis
+func (c *Client) Set(ctx context.Context, s *session.Session) error {
 	if s == nil {
-		log.Event(ctx, "session is empty", log.ERROR)
-		return errors.New("session is empty")
+		log.Event(ctx, ErrEmptySession, log.ERROR)
+		return errors.New(ErrEmptySession)
 	}
 
-	json, err := s.MarshalJSON()
+	sJSON, err := s.MarshalJSON()
 	if err != nil {
-		log.Event(ctx, "failed to marshal session", log.Error(err), log.ERROR)
+		log.Event(ctx, ErrFailedToMarshal, log.Error(err), log.ERROR)
 		return err
 	}
 
-	msg, err := c.client.Set(s.ID, json, c.ttl).Result()
+	msg, err := c.client.Set(s.ID, sJSON, c.ttl).Result()
 	if err != nil {
 		log.Event(ctx, msg, log.Error(err), log.ERROR)
+		return err
 	}
-	log.Event(ctx, msg, log.INFO)
 
 	return nil
 }
+
+// GetByID - gets a session from redis using its ID
+func (c *Client) GetByID(ctx context.Context, id string) (*session.Session, error) {
+	if id == "" {
+		log.Event(ctx, ErrEmptySessionID, log.ERROR)
```

Comment:
generally speaking you should either log or handle the error. In this case I would remove the log and just return it. Otherwise you end up with multiple points repeating the same error.